### PR TITLE
core - enable richer policy metadata

### PR DESCRIPTION
**Pull request by JohnHillegass**
For #6445, enables richer policy metadata by allowing object and array types in the schema validation.

### CODE REVIEW

**Comment by JohnHillegass** [comment]
I had considered adding an any_dict for this but thought this may read be clearer

```
        'any_dict': {
            "type": "object",
            "patternProperties": {
                "": {
                    'oneOf': [
                        {"type": "string"},
                        {"type": "boolean"},
                        {"type": "number"},
                        {"type": "object"},
                        {"type": "array"},
                    ],
                }
            },
        },
```

File: c7n/schema.py [hunk 1]
```diff
@@ -258,7 +258,7 @@ def generate(resource_types=()):
                 'comments': {'type': 'string'},
                 'description': {'type': 'string'},
                 'tags': {'type': 'array', 'items': {'type': 'string'}},
-                'metadata': {'$ref': '#/definitions/basic_dict'},
+                'metadata': {'type': ['object', 'array']},
```
Comment (kapilt) [inline c7n/schema.py:261]: just object, people can put an array under a key, also jsonschema that would be a oneOf / anyOf block to support both afaik

Comment (JohnHillegass) [reply c7n/schema.py]: Ok cool, sounds good. Just made the update in 9753620

Comment (JohnHillegass) [reply c7n/schema.py]: Heads up, I also needed to re-record an IAM test in 38ddfaa to pass CI. Kind of weird it randomly stopped working but seems to be fine now 🤷.

Comment (JohnHillegass) [reply c7n/schema.py]: Hmm, actually that test seems to be a timebomb that will break every 30 days. Rethinking what the fix should be 🤔

Comment (JohnHillegass) [reply c7n/schema.py]: Ok, d044909 should do it.

Comment (kapilt) [reply c7n/schema.py]: thanks! re flakey

**Review by kapilt — APPROVED** [review]
thanks, lgtm

**Comment by kapilt** [comment]
prioritizing a merge here cause its fixing a block against all extants prs, thanks @JohnHillegass

**Pull request merged**